### PR TITLE
[Feature] Use CAS for All Authentication

### PR DESCRIPTION
**Pull request by cslzchen**
### Ticket:

https://openscience.atlassian.net/browse/OSF-6438
https://openscience.atlassian.net/browse/SEC-26
https://openscience.atlassian.net/browse/OSF-6450
### Changes:
- Remove auto login from OSF form submission
- Generate credentials from `(username, verification_key)` to log user back in OSF after:
  - reset password through "forgot password"
  - set password when added as an unregistered contributor
  - create account
- Improve CAS Login Page
### Side Effects:

Must be deployed together with the [OSF-PR#5804](https://github.com/CenterForOpenScience/osf.io/pull/5804).

### CODE REVIEW

File: cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java [hunk 1]
```diff
@@ -275,6 +275,17 @@ protected Credential constructCredentialsFromRequest(final RequestContext contex
 
             return credential;
         }
+        // Construct credential if presented with (username, verification_key)
+        // This is used when:
+        //      1. User creates an account
+        //      2. User resets the password through forgot_password
+        //      3. User sets password when added as an unregistered contributor
+        //      4. Other situations
+        else if (request.getParameter("verification_key") != null) {
```
Comment (icereval) [inline cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java]: Should also add a check for username here.

File: etc/cas.properties [hunk 2]
```diff
@@ -38,6 +38,7 @@ osf.url=http://localhost:5000/
 osf.institutionLogin.url=http://localhost:5000/login/?campaign=institution
 osf.resendConfirmation.url=http://localhost:5000/resend/
 osf.forgotPassword.url=http://localhost:5000/forgotpassword/
+osf.createAccount.url=http://localhost:5000/login/
```
Comment (icereval) [inline etc/cas.properties]: Just a reminder for deployment, we'll need to add this in the stage/prod configs.

**Comment by icereval** [comment]
Looks good!, first pass finished.

File: etc/cas.properties [hunk 3]
```diff
@@ -38,6 +38,7 @@ osf.url=http://localhost:5000/
 osf.institutionLogin.url=http://localhost:5000/login/?campaign=institution
 osf.resendConfirmation.url=http://localhost:5000/resend/
 osf.forgotPassword.url=http://localhost:5000/forgotpassword/
+osf.createAccount.url=http://localhost:5000/register/
```
Comment (cslzchen) [inline etc/cas.properties:41]: Reminder: update `cas.properties` for deployment. Sign-up page now uses `/register` instead of `/login`.

**Pull request merged**